osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -203,8 +203,8 @@ private void editRequested(BeatmapInfo beatmap)
             Push(new Editor());
         }
 
-        private void onBeatmapRestored(BeatmapInfo b) => carousel.UpdateBeatmap(b);
-        private void onBeatmapHidden(BeatmapInfo b) => carousel.UpdateBeatmap(b);
+        private void onBeatmapRestored(BeatmapInfo b) => Schedule(() => carousel.UpdateBeatmap(b));

```

Comment:
I don't think the schedule call should be at this place, but rather at the caller's place unless you explicitly mark these methods (via xml-doc) as being thread-safe.

Reply:
Considered this, but the caller doesn't have access to a scheduler as they are not a drawable.